backend/product-catalog-service/src/main/java/com/furniture_store/product_catalog/entity/Product.java

```diff
@@ -0,0 +1,38 @@
+package com.furniture_store.product_catalog.entity;
+
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+@Table
+@Entity
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class Product {
```

Comment:
The use of Lombok's @Builder annotation on JPA @Entity classes in Java can lead to problems for several reasons:

1. There is a risk of manually passing the id value, which will lead to conflicts with autoincrement strategies or duplicate identifiers.

2. For @Entity objects, this can lead to the creation of unnecessary copies of the objects, which then have to be stored, updated or deleted from the database.

3. Using @Builder can confuse other developers who may not expect that these objects are created outside the JPA context.

https://thorben-janssen.com/lombok-hibernate-how-to-avoid-common-pitfalls/
https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/